brave-spring-web-servlet-interceptor/src/test/java/com/github/kristofa/brave/spring/ITServletHandlerInterceptor.java

```diff
@@ -0,0 +1,70 @@
+package com.github.kristofa.brave.spring;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.http.ITServletContainer;
+import com.github.kristofa.brave.http.SpanNameProvider;
+import java.io.IOException;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.context.support.AnnotationConfigWebApplicationContext;
+import org.springframework.web.servlet.DispatcherServlet;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+public class ITServletHandlerInterceptor extends ITServletContainer {
+
+  @Controller
+  static class TestController {
+
+    @RequestMapping(value = "/foo")
+    public ResponseEntity<Void> foo() throws IOException {
+      return ResponseEntity.ok().build();
+    }
+
+    @RequestMapping(value = "/disconnect")
+    public ResponseEntity<Void> disconnect() throws IOException {
+      throw new IOException();
+    }
+  }
+
+  @Configuration
+  @EnableWebMvc
+  @Import(ServletHandlerInterceptor.class)
+  static class TracingConfig extends WebMvcConfigurerAdapter {
+
+    @Autowired
+    private ServletHandlerInterceptor interceptor;
+
+    @Override
+    public void addInterceptors(InterceptorRegistry registry) {
+      registry.addInterceptor(interceptor);
+    }
+  }
+
+  @Override protected void init(ServletContextHandler handler, Brave brave,
+      SpanNameProvider spanNameProvider) {
+
+    AnnotationConfigWebApplicationContext appContext =
```

Comment:
this is tricky.. but removes terror of using static references to pass brave inside a configuration class.